internal/token_test.go

```diff
@@ -265,23 +265,23 @@ func TestFrontmatter(t *testing.T) {
 			[]TokenType{FrontmatterFenceToken, TextToken, FrontmatterFenceToken, TextToken, StartTagToken, TextToken, EndTagToken, TextToken},
 		},
 		{
-			"tokenizes elements inside",
+			"does not tokenize elements inside",
 			`
 			---
 			const a = <div />;
 			---
 			`,
-			[]TokenType{FrontmatterFenceToken, TextToken, SelfClosingTagToken, TextToken, FrontmatterFenceToken},
+			[]TokenType{FrontmatterFenceToken, TextToken, TextToken, FrontmatterFenceToken},
 		},
 		{
-			"elements can have expression as child in frontmatter",
+			"no elements or expressions in frontmatter",
 			`
 			---
 			const contents = "foo";
 			const a = <div>{contents}</div>;
 			---
 			`,
-			[]TokenType{FrontmatterFenceToken, TextToken, TextToken, StartTagToken, StartExpressionToken, TextToken, EndExpressionToken, EndTagToken, TextToken, FrontmatterFenceToken},
+			[]TokenType{FrontmatterFenceToken, TextToken, TextToken, TextToken, TextToken, TextToken, TextToken, TextToken, FrontmatterFenceToken},
```

Comment:
Yeah to your PR comment, not important for this PR at all, but we could probably collapse all the text tokens into one now.

Reply:
The parser collapses any adjacent TextTokens into a single TextNode, so I'm not sure if we need to tackle this in the tokenizer.